frontend/src/queries/Project.ts

```diff
@@ -58,3 +91,88 @@ export function useCreateProjectMutation(): UseMutationReturnType<
         },
     });
 }
+
+export function useUpdateProjectMutation(): UseMutationReturnType<
+    void,
+    Error,
+    { projectId: number; projectData: Partial<ProjectForm> },
+    void
+> {
+    const queryClient = useQueryClient();
+    return useMutation<void, Error, { projectId: number; projectData: Partial<ProjectForm> }, void>(
+        {
+            mutationFn: ({ projectId, projectData }) => updateProject(projectId, projectData),
+
+            onSuccess: (_, variables) => {
+                queryClient.invalidateQueries(PROJECT_QUERY_KEY(variables.projectId));
+                console.log("Project updated successfully.");
+            },
+
+            onError: (error) => {
+                console.error("Project update failed", error);
+                alert("Failed to update project. Please try again.");
+            },
+        }
+    );
+}
```

Comment:
```suggestion
export function useUpdateProjectMutation(): UseMutationReturnType<
    Project,
    Error,
    { projectId: number; projectData: Partial<ProjectForm> },
    void
> {
    const queryClient = useQueryClient();
    return useMutation<Project, Error, { projectId: number; projectData: Partial<ProjectForm> }, void>(
        {
            mutationFn: ({ projectId, projectData }) => updateProject(projectId, projectData),

            onSuccess: (_, variables) => {
                queryClient.invalidateQueries({ queryKey: PROJECT_QUERY_KEY(variables.projectId) });
                console.log("Project updated successfully.");
            },

            onError: (error) => {
                console.error("Project update failed", error);
                alert("Failed to update project. Please try again.");
            },
        }
    );
}
```

Reply:
De API geeft het geüpdatete project terug, dus kan je dat evengoed weerspiegelen in de query. Als iemand later ooit wel dat project nodig heeft, kan die dat gebruiken.

Reply:
ja khebt gdn